### PR TITLE
Resque instrumentation (WIP)

### PR DESCRIPTION
**Pull request by mikker**
Closes elastic/apm-agent-ruby#75

### CODE REVIEW

**Comment by mikker** [comment]
@simitt I'm having trouble again – in the forked version, Spans aren't added to the Transaction. Can you see what the trouble might be?

**Comment by mikker** [comment]
Thinking more about I think the problem might be that the spans are added in the forked version of the transaction and therefore not brought back after the job has run. Can't see a solution to this right now other than going back to some kind of inter-process comm. but will think some more.

**Comment by simitt** [comment]
Yes I am pretty sure the issue is that the transaction is started in another process. I didn't have time to look into it again yet, and probably will be busy with something else a view days. 
We probably could look into starting the transaction+spans inside the child process but yielding it back to the parent process before the child is stopped.

**Comment by simitt** [comment]
As just discussed over zoom with @mikker, I did a little POC for using `IO.pipe` for inter-process communication, see https://github.com/simitt/apm-agent-ruby/commit/96bcea5b0c9dada30f523b500311bd1f9139d24c

Using pipes seems relatively straight forward code wise. 
However, it is pretty easy to block the reading thread by not sending anything to the pipe or by not using a delimiter `\n` as the reader will then wait forever. 
As a `transaction` cannot simply be dumped with `Marshal.dump`, manual (de-)serialization is necessary. In the POC I use the existing serialization code to enable dumping the transaction. However, for loading the transaction additional code for manual deserialization would be necessary. Or, as an alternative the serialized transaction could be added as is to the transactions array that gets sent to the Server.

To summarize, using `IO.pipe` looks like a good concept for handling interprocess communication, but as neither @mikker nor I have experience with possible pitfalls of using `IO.pipe` and quite some additional tests would be required to make the code resilient, it might be a better option to implement a synchronous call to the APM Server per Resque job for now and revisit after going GA if needed.
